app/src/lib/promise.ts

```diff
@@ -7,35 +7,13 @@
  *
  *
  * @param action the promise work to track
- * @param timeout the minimum time to wait before resolving the promise (in milliseconds)
+ * @param timeoutMs the minimum time to wait before resolving the promise (in milliseconds)
  */
 export function promiseWithMinimumTimeout<T>(
   action: () => Promise<T>,
-  timeout: number
+  timeoutMs: number
 ): Promise<T> {
-  return new Promise<T>((resolve, reject) => {
-    let timeoutExpired = false
-    let result: T | null = null
-
-    const resolveIfBothDone = () => {
-      if (result != null && timeoutExpired) {
```

Comment:
Looking at this diff now I'm pretty sure this is a bug and if the `action` returns `null` or `undefined` then `promiseWithMinimumTimeout` will never resolve. Gonna add a test to confirm

Reply:
You're right!  🙈